inference-engine/include/details/ie_mmap_allocator.hpp

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief The header file defines utility PreAllocator class
+ * 
+ * @file ie_pre_allocator.hpp
+ */
+#pragma once
+
+#include <details/ie_exception.hpp>
+#include <memory>
+
+#include "ie_allocator.hpp"
+
+#ifdef _WIN32
+#include <windows.h>
+#define MAP_FAILED nullptr
+#else
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#define INVALID_HANDLE_VALUE -1
```

Comment:
I think, it better to use constant expressions inside appropriate namespace (instead defines in common namespace), otherwise we can clash with names in client applications.